src/main/java/baseball/controller/GameManager.java

```diff
@@ -0,0 +1,62 @@
+package baseball.controller;
+
+import baseball.io.PlayerIoManager;
+import baseball.model.ball.Ball;
+import baseball.model.ball.BallMaker;
+import baseball.model.judge.BaseballJudge;
+import baseball.model.judge.JudgeResult;
+import java.util.List;
+
+public class GameManager {
+
+    private PlayerIoManager playerIoManager;
+    private BallMaker ballMaker;
+    private BaseballJudge baseballJudge;
+
+    private GameManager() {
+    }
+
+    public static GameManager create() {
+        GameManager gameManager = new GameManager();
+
+        gameManager.playerIoManager = PlayerIoManager.create();
+        gameManager.ballMaker = new BallMaker();
+
+        return gameManager;
+    }
```

Comment:
static 메서드를 사용하는 것보다 생성자를 활용해 보는 건 어떤가요?

`create()`가 담당하는 역할이 GameManager 객체를 생성할 때 관련된 의존성을 주입하는 것이라 생각해요!
객체를 생성할 때 의존성을 주입하는 것은 어떤가요? DI의 생성자 주입과 관련되어 있어요! [(참고)](https://mangkyu.tistory.com/125)
DI를 적용하면 코드가 훨씬 단순해져서 가독성이 좋아질 거라고 생각합니다!

```suggestion
public GameManager() {
        this.playerIoManager = new PlayerIoManager(); 
        this.ballMaker = new BallMaker();
    }
```

Reply:
안녕하세요 시연님! 백엔드 파트로 지원한 최승준이라고 합니다! 제가 공부한 것과 다른 내용이 있어 의견을 남깁니다!
제가 아는 바로는 생성자 주입은 생성자를 통해 객체에 필요한 의존성을 주입받는 방식으로 알고 있으며,
생성자 메서드의 매개변수를 통해 필요한 의존성을 입력받는 것으로 공부하였는데요!
생성자 메서드 내부에서 new를 통해 객체를 생성하여 필드값을 초기화하는 것은 생성자 주입과는 조금 다른 결이 아닐까 조심스레 의견 남겨봅니다!

Reply:
말씀하신것과 같이

- 객체를 생성하고 나서 의존성을 주입하는 것
- 객체 생성 시점에 의존성을 주입하는 것

위 두 개는 분명히 차이점을 갖는군요! 저는 사실 의존성 주입하면 생성자 주입만 생각했었는데, 덕분에 인사이트를 얻었습니다. 감사합니다 :D

그럼 만약에, 이렇게 생성자 내부에서 직접 인스턴스를 생성해서 의존성을 할당하지 않고, 생성자 주입을 사용하게 되는 경우는 main 메서드에서 무수히 많은 인스턴스를 미리 선언해둬야할 것 같은데 이 부분에 대해서는 어떻게 생각하시는지 시연님의 의견이 궁금합니다 :D